packages/experimental/Button/src/CompoundButton/CompoundButton.tsx

```diff
@@ -0,0 +1,52 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { Image, View } from 'react-native';
+import { CompoundButtonProps, compoundButtonName, CompoundButtonType } from './CompoundButton.types';
+import { Text } from '@fluentui-react-native/experimental-text';
+import { stylingSettings } from './CompoundButton.styling';
+import { filterImageProps } from '@fluentui-react-native/adapters';
+import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
+import { useButton } from '../useButton';
+
+export const CompoundButton = compose<CompoundButtonType>({
+  displayName: compoundButtonName,
+  ...stylingSettings,
+  slots: {
+    root: View,
+    icon: Image,
+    content: Text,
+    secondaryContent: Text,
+    contentContainer: View,
+  },
+  filters: {
+    icon: filterImageProps,
+  },
+  render: (userProps: CompoundButtonProps, useSlots: UseSlots<CompoundButtonType>) => {
+    const { icon, content, secondaryContent, ...rest } = userProps;
```

Comment:
I wonder if these should be split out in the returned closure. By that I mean something like:

```ts
const { icon, content, secondaryContent, ...merged } = mergeProps(button.props, final);
```

Reply:
This is to ensure it can be set by a component in JSX if this component is being driven as part of a larger component.